features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/callback/RecorderWindowCallback.kt

```diff
@@ -51,13 +53,15 @@ internal class RecorderWindowCallback(
     @MainThread
     override fun dispatchTouchEvent(event: MotionEvent?): Boolean {
         if (event != null) {
-            // we copy it and delegate it to the gesture detector for analysis
-            @Suppress("UnsafeThirdPartyFunctionCall") // internal safe call
-            val copy = copyEvent(event)
-            try {
-                handleEvent(copy)
-            } finally {
-                copy.recycle()
+            if (touchPrivacy == TouchPrivacy.SHOW) {
```

Comment:
this API is only set in the configuration so it is immutable. I think instead we should make sure we never register a touch recorder in case this is `HIDE`. There's no interest in keeping it and is better optimised this way.